File: Algo/DynamicProgramming/TakeYouForward/practice/tuf/Sorting/SelectionSort.java
package tuf.Sorting;

import java.util.Arrays;
import java.util.stream.Collectors;

public class SelectionSort {
    int[] arr ={4,2,0,1,23,45234,-12,23343};
    public static void main(String[] args) {
        new SelectionSort();
    }

    SelectionSort(){
        sort(arr);
        print(arr);
    }

     public void sort(int[] arr){
        int N= arr.length;
        for(int i=0;i<N;i++){
            int mini=i;
            for(int j=i;j<N;j++){
                if(arr[j]<arr[mini]) mini=j;
            }
            swap(arr,i,mini);
        }
    }

    public void swap(int[] arr,int src,int dest){
        int temp = arr[src];
        arr[src]= arr[dest];
        arr[dest]=temp;

}

    public void print(int[] arr){
        System.out.println(Arrays.stream(arr).boxed().collect(Collectors.toList()));
    }
}

File: Algo/DynamicProgramming/TakeYouForward/practice/practice3/Sorting/selectionsort.go
package main

import "fmt"

func swap2(src int, dest int, arr [8]int) [8]int {
	temp := arr[src]
	arr[src] = arr[dest]
	arr[dest] = temp
	return arr
}

func main2() {
	fmt.Print(" Selection Sort")
	array := [8]int{8, 4, 2, 5, 1, 6, 3, 9}
	n := len(array)

	for i := 0; i < n; i++ {
		max := i
		for j := i; j < n; j++ {
			if array[j] > array[max] {
				max = j
			}
		}
		array = swap(max, i, array)
	}
	fmt.Print(array)
}
